extensions/amp-story/1.0/progress-bar.js

```diff
@@ -423,6 +431,49 @@ export class ProgressBar {
     }
   }
 
+  /**
+   * Show/hide ad progress bar treatment based on ad visibility.
+   * @param {boolean} adState
+   */
+  onAdStateUpdate_(adState) {
+    // TODO(ccordry): follow up with branched experiment for different
+    // time intervals.
+    if (!isExperimentOn(this.win_, 'story-ad-progress-segment')) {
+      return;
+    }
+    // Set CSS signal that we are in the experiment.
+    // TODO(#33969) delete when launched.
+    if (!this.root_.hasAttribute('i-amphtml-yellow-progress')) {
```

Comment:
Nit: plz don't tie what it looks like with the name of the class, if we change it to red tomorrow we don't want to update the classnames

Reply:
done.